### PR TITLE
635- Dynamic display continuous measurement points

### PR DESCRIPTION
**Pull request by anaislet-c2c**
### JIRA issue

https://jira.camptocamp.com/browse/GSREN3D-635
https://jira.camptocamp.com/browse/GSREN3D-614

### Description

- change style of points on hover and click
- link checkbox to layer displayed
- add spot-measurement layer (WARNING this is a fake layer waiting the real data)

### Screenshots

https://github.com/sigrennesmetropole/cooperation_jn_wave/assets/126822753/17d8a738-62ef-422f-97fa-95defaf834c9

### CODE REVIEW

File: src/interactions/clickAndMoveInteraction.ts [hunk 1]
```diff
@@ -11,51 +11,104 @@ import type { RennesApp } from '@/services/RennesApp'
 import { usePointsStore } from '@/stores/points'
 import { RENNES_LAYER } from '@/stores/layers'
 import { useViewsStore } from '@/stores/views'
+import {
+  getUnselectedPointStyle,
+  getSelectedPointStyle,
+} from '../../src/style/common'
```
Comment (ismailsunni) [inline src/interactions/clickAndMoveInteraction.ts:17]: change to `../style/common`

File: src/stores/checkbox.ts [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+      layerStore.enableLayer(RENNES_LAYER.customLayerSpotData)
+    } else {
+      layerStore.disableLayer(RENNES_LAYER.customLayerSpotData)
+    }
+  }
+
+  function toggleEmitterSitesPoint() {
+    selectedPointEmitterSites.value = !selectedPointEmitterSites.value
+    if (selectedPointEmitterSites.value == true) {
+      // layerStore.enableLayer(RENNES_LAYER.customLayerEmitterSites)
```
Comment (ismailsunni) [inline src/stores/checkbox.ts:33]: Is this for later?

File: src/style/common.ts [hunk 3]
```diff
@@ -2,11 +2,34 @@
...
 
-export const ContinuousMeasurementStyle: Style = new Style({
-  image: new Icon({
-    opacity: 1,
-    src: continuousIcon,
-    scale: 1,
-  }),
-})
+function getIconFromLayerName(layerName) {
+  if (layerName === RENNES_LAYER.customLayerContinuousMeasurement) {
```
Comment (ismailsunni) [inline src/style/common.ts:10]: It's also good to put them in a dictionary `layerName` as the key, and the icon as the value. But this is also fine. Perhaps after having more than 2, we can switch to a dictionary for easier reading

**Review by ismailsunni — APPROVED** [review]
LGTM

File: src/stores/checkbox.ts (hunk 2 above)
Comment (anaislet-c2c) [reply src/stores/checkbox.ts:33]: Yes :-)

**Pull request merged**